lib/models/candle/sync_range.js

```diff
@@ -1,18 +1,89 @@
-'use strict'
-
-const debug = require('debug')('bfx:hf:models:candle:sync')
-const genSyncRange = require('../../util/gen_sync_range')
-const { candleWidth } = require('bfx-hf-util')
-
-/**
- * @param {RESTv2} rest
- * @param {Object} args
- * @param {boolean?} args.forceSync - if true, no gap filling is performed
- * @param {string} args.symbol
- * @param {string} args.tf
- * @param {number} args.start
- * @param {number} args.end
- */
-module.exports = genSyncRange('candles', ({ tf }) => {
-  return candleWidth(tf) * 2 // for maxDiscrepancy
-}, ['symbol', 'tf'], debug)
+const DEBUG_STR = 'bfx:hf:models:candle:sync-range' // for pretty formatting
+
+const debug = require('debug')(DEBUG_STR)
+const { RESTv2 } = require('bfx-api-node-rest')
+const PromiseThrottle = require('promise-throttle')
+const PI = require('p-iteration')
+const _last = require('lodash/last')
+const Promise = require('bluebird')
+
+const FETCH_LIMIT = 1000
+const rest = new RESTv2({ transform: true })
+const pt = new PromiseThrottle({
```

Comment:
Nice, I wonder if we should add an exponential backoff retry just incase we do get rate limited?